Makefile

```diff
@@ -0,0 +1,40 @@
+# stop npm installing for every locally run command but still being a dependency before they can be run first
+INSTALL_STAMP = .npm_installed
+# docker build as dependency but only build it the first itme
+IMAGE_STAMP = .image_built
+
+# headless run
+test: install
+	npx playwright test
+
+# run with browser open
+debug: install
+	npx playwright test --debug
+
+install: $(INSTALL_STAMP)
+
+$(INSTALL_STAMP): package.json package-lock.json
+	npm install
+	npx playwright install
+	npm install --save-dev prettier husky lint-staged
+	npx husky install
+	touch $(INSTALL_STAMP)
+
+lint:
```

Comment:
Could we make this run via docker only? That way we shouldn't require npm being a required local dependency?

I'm thinking to run this via an image that has npm already and just mount the whole root of the project to it and that way we run the linter inside docker too

Reply:
as discussed, for now, I think it's simplest to keep the docker stuff separate as is due to debugging requiring a local install but we can keep this in mind